Sources/TootSDK/TootClient/TootClient.swift

```diff
@@ -160,10 +160,16 @@ extension TootClient {
 
     /// Fetch data asynchronously and return the raw response.
     internal func fetch(req: HTTPRequestBuilder) async throws -> (Data, HTTPURLResponse) {
-        if req.headers.index(forKey: "Content-Type") == nil {
+        if req.headers.index(forKey: "Content-Type") == nil && req.body != nil {
```

Comment:
I would prefer if the changes to the request are specific to the flavour which mandates them. In this case, if flavour is sharkey, then we apply the content type modifications.

Not all flavours handle this in the same way. For example, Pleroma/Akkoma would always look for the content-type header and default to "application/json" if it's not set. So the behaviour and compatibility may vary. It will be best to make such instances explicit.